gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -440,9 +440,20 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) (err error) {
 	if !strings.HasPrefix(pkgPath, chainDomain+"/") {
 		return ErrInvalidPkgPath("invalid domain: " + pkgPath)
 	}
+
+	// Extra keeper-only checks.
+	gm, err := gnomod.ParseMemPackage(memPkg)
+	if err != nil {
+		return ErrInvalidPackage(err.Error())
+	}
+
 	if pv := gnostore.GetPackage(pkgPath, false); pv != nil {
-		return ErrPkgAlreadyExists("package already exists: " + pkgPath)
+		// Private packages can be re-uploaded (overwritten).
+		if !gm.Private {
+			return ErrPkgAlreadyExists("package already exists: " + pkgPath)
+		}
```

Comment:
Should we remove object of the previous realm from the store ?

Reply:
think we should do with the package value as well as the memPackage. i think we can make it in another PR, that it's necessary to do but not necessarily belong to the scope of this PR.